### PR TITLE
docs: support session replay tunnelling for javascript

### PR DESCRIPTION
**Pull request by rhuanbarreto**
## DESCRIBE YOUR PR
*Tell us what you're changing and why. If your PR **resolves an issue**, please link it so it closes automatically.*

If a session replay is embedded into the envelope, decoding it to text will yield a `400 Bad Request` when tunnelling. So we need to pass the content as bytes into the body in order to work.

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [X] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## LEGAL BOILERPLATE

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: docs/platforms/javascript/common/troubleshooting/index.mdx [hunk 1]
```diff
@@ -221,7 +221,8 @@ const SENTRY_PROJECT_IDS = ["123456"];
 
 export const postAction = async ({ request }) => {
   try {
-    const envelope = await request.text();
+    const envelopeBytes = await req.arrayBuffer();
+    const envelope = new TextDecoder().decode(envelopeBytes);
```
Comment (bruno-garcia) [inline docs/platforms/javascript/common/troubleshooting/index.mdx:225]: Looks like `TextDecoder` anyway[ expects UTF-8 or other type string](https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder). What's the advantage from calling `.text` on request?

**Review by billyvg — APPROVED** [review]
This seems correct, haven't tested it though

File: docs/platforms/javascript/common/troubleshooting/index.mdx (hunk 1 above)
Comment (rhuanbarreto) [reply docs/platforms/javascript/common/troubleshooting/index.mdx:225]: There are 2 explanations:

- Session replays are sent as compressed zip (binary) appended at the end of the envelope body. When you get the content of the session replay as text you lose data in the transfer
- if you use `.text()` then the body was used and you can't reuse it from the request to get its bytes

Therefore I get the bytes and save it separately to forward to Sentry and then I convert the bytes to text to extract the textual part of it to validate.

This is already validated and running in production for me. Works very well.

**Comment by rhuanbarreto** [comment]
The CI failure is unrelated to this PR change.

**Review by bruno-garcia — APPROVED** [review]
Thanks @rhuanbarreto

**Pull request merged**